nomenklatura/db.py

```diff
@@ -23,11 +31,31 @@ def get_metadata() -> MetaData:
     return MetaData()
 
 
+@cache
+def get_statement_table() -> Table:
+    return make_statement_table(get_metadata(), DB_STORE_TABLE)
+
+
 @contextmanager
 def ensure_tx(conn: Connish = None) -> Generator[Connection, None, None]:
-    if conn is not None:
-        yield conn
-        return
-    engine = get_engine()
-    with engine.begin() as conn:
-        yield conn
+    try:
+        if conn is not None:
+            yield conn
+        else:
+            engine = get_engine()
+            with engine.begin() as conn:
+                yield conn
+    finally:
+        if conn is not None:
```

Comment:
Wait, are we committing after an exception here? That feels a bit unhealty.

Reply:
Fixed it here: https://github.com/opensanctions/nomenklatura/pull/125/commits/f968f9087caac2ea570a520cb4c97374fb75da28 and here: https://github.com/opensanctions/nomenklatura/pull/125/commits/238909462b8adfebc1f80ea0755c47b3e9cb9d03